src/main/java/net/dries007/tfc/client/render/ChestItemRenderer.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.client.render;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.BlockEntityWithoutLevelRenderer;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.client.renderer.block.model.ItemTransforms;
+import net.minecraft.client.renderer.blockentity.BlockEntityRenderDispatcher;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.block.Block;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import net.dries007.tfc.common.blockentities.TFCChestBlockEntity;
+import net.dries007.tfc.common.blockentities.TFCTrappedChestBlockEntity;
+import net.dries007.tfc.common.blocks.wood.TFCTrappedChestBlock;
+
+public class ChestItemRenderer extends BlockEntityWithoutLevelRenderer
+{
+    private final TFCChestBlockEntity be;
+    private final BlockEntityRenderDispatcher dispatch;
+
+    public ChestItemRenderer(Block block)
+    {
+        super(Minecraft.getInstance().getBlockEntityRenderDispatcher(), Minecraft.getInstance().getEntityModels());
+        if (block instanceof TFCTrappedChestBlock)
+        {
+            be = new TFCTrappedChestBlockEntity(BlockPos.ZERO, block.defaultBlockState());
+        }
+        else
+        {
+            be = new TFCChestBlockEntity(BlockPos.ZERO, block.defaultBlockState());
+        }
```

Comment:
can you not do `block.newBlockEntity(...)`? That would remove the switch here. (Also `be` is a rubbish name `chest` would've been better but heck)